include/PathSimulator.hpp

```diff
@@ -0,0 +1,140 @@
+#ifndef DDSIM_PATHSIMULATOR_HPP
+#define DDSIM_PATHSIMULATOR_HPP
+
+#include "CircuitOptimizer.hpp"
+#include "CircuitSimulator.hpp"
+
+#include <future>
+#include <limits>
+#include <taskflow/taskflow.hpp>
+#include <thread>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <variant>
+#include <vector>
+
+class PathSimulator: public CircuitSimulator {
+public:
+    // Add new strategies here
+    enum class Mode {
+        Sequential,
+        SequentialFalse,
+        PairwiseRecursiveGrouping,
+        BracketGrouping3,
+        BracketGrouping7,
+        BestCase,
```

Comment:
It's probably best to also rename the `BestCase` strategy to something more suitable for general use. In the end, it is a strategy that, given a circuit that actually consists of two circuits,  starts in the middle of both circuits and alternates between applications of gates from either circuit. 

Maybe it would be a good idea to change the strategy in a way that it just takes a parameter that indicates the middle of the circuit from which to start from. This parameter can be defaulted to half the number of gates.

Reply:
strategie is currently implemented directly in python. Should this change?

Reply:
yeah. it would probably make sense to move the strategy to the C++ part for consistency. As stated in the above comment it just takes a parameter that indicates the starting point of the simulation path and then alternates between left and right applications. you could even call the corresponding mode `Alternating`